pkg/kube/inject/webhook.go

```diff
@@ -1075,13 +1078,16 @@ func (wh *Webhook) inject(ar *kube.AdmissionReview, path string) *kube.Admission
 		injectedAnnotations: wh.Config.InjectedAnnotations,
 		proxyEnvs:           parseInjectEnvs(path),
 	}
-	clusterID, _ := extractClusterAndNetwork(params)
-	if wh.namespaces != nil {
-		client := wh.namespaces.ForCluster(cluster.ID(clusterID))
+
+	if openShiftNamespaceTest != nil {
+		// Mocked namespace to cover unit tests
```

Comment:
This seems unexpected. Why can't we just pass in a `p.MultiCluster` that has the mocks we need?